_posts/2015-03-stitch-fix-heart-unix.md

```diff
@@ -0,0 +1,139 @@
+   - [Can I efficiently and securely copy these files?]()
+ - Directories
+   - [Directory File Summary]()
+   - [Sort Directories By Number Of Files]()
+   - [Delete Directory With Large Number Of Files]()
+ - CSVs
+   - [Filter CSV File By Column Values]()
+   - [Index CSV File]()
+ - Systems 
+   - [Display Resource Usage and Availability]()
+   - [Easily Manage Python Virtual Environments]()
+ - Productivity
+   - [Alias Everything]()
+   - [Command Completion Alert]()
+ - [Bemuse Coworkers]()
+
+<a name="check-image-size"></a>
+### Check Image Size
+&#8212; [Dave Copeland](https://twitter.com/davetron5000/) *(OS X / bash)*
+
+#### Problem
+Check the sizes of our product images the pickers use to check what they are picking in our warehouse
+
+#### Solution
+Created a script to run curl on the argument to download the image, use ImageMagick to get its size and print that out in CSV.  Piped my input CSV of images into `xargs -n1 -P8 ./my_script.rb` to basically run my script 8-way parallel to get the job done as fast as I could without setting my machine on fire.
+
+##### Input
+Simple CSV of some ids and an image URL
+
+##### Command
+```
+cat my_input.csv | xargs -n1 -P8 ./check.rb > images_with_possible_issues.csv
+```
+
+ - `xargs` runs a command, feeding `xargs` `STDIN` as arguments to that command
+   - `echo ""foo"" | xargs ls` is the same as `""ls foo""`.  
+ - `-n1` says to run the given command once for each line of input.  
+  - Normally `xargs` will run many lines at once, so if you had a file with 10 rows in it called `""blah.csv""` and do `cat blah.csv | xargs curl`, it would likely run curl once with all 10 rows of blah.csv given to curl.  `-n1` means to just do them one at a time (so 10 runs of curl in this example).
+ - `-P8` says to parallelize it 8 ways.  
+
+So, I've got 8 instances of my script running at once.  Obviously, there are diminishing returns on parallelism, but since curl'ing images is mostly I/O bound this worked pretty well without compromising my machine.
+
+### Find Recently Changed Files
+&#8212; [Greg Novak](https://www.linkedin.com/in/gsnovak/) *(OS X / bash)*
+
+Some file changed but I don't know where.  Sometimes this is "I don't know where my web browser saved my file."  I want a list of the most recently changed file in a tree
+
+### Count Data Dump Number Of Lines
+&#8212; [Deep Ganguli](https://twitter.com/dgangul1/) *(OS X / bash)*
+
+How many lines of text are there in a data dump? This happens all the time, and I hate opening the data in a text editor and scrolling to the bottom.
+
+### Bulk Change Filenames
+&#8212; [Greg Novak](https://www.linkedin.com/in/gsnovak/) *(OS X / bash)*
+
+I have a bunch of files and I want to change all of their names at once.
+
+### Efficient Secure File Copy
+&#8212; [Greg Novak](https://www.linkedin.com/in/gsnovak/) *(OS X / bash)*
+
+"I'm sick of typing:
+scp -i ~/path/to/pem/file.pem some-file.txt ubuntu@ec2-11-22-33-44.amazon.com:/home/ubuntu/some/path
```

Comment:
Looks like @gnovak made it up:

`ec2-11-22-33-44.amazon.com`

1
2
3
4
:smile: